### PR TITLE
devfs and ufs changes

### PR DESCRIPTION
**Pull request by qwattash**

### CODE REVIEW

File: sys/ufs/ffs/ffs_softdep.c [hunk 1]
```diff
@@ -6754,14 +6755,13 @@ complete_trunc_indir(freework)
 	/*
 	 * Zero the pointers in the saved copy.
 	 */
+	start = (uintptr_t)indirdep->ir_savebp->b_data;
 	if (indirdep->ir_state & UFS1FMT)
-		start = sizeof(ufs1_daddr_t);
+		len = sizeof(ufs1_daddr_t) * freework->fw_start;
```
Comment (jrtc27) [inline sys/ufs/ffs/ffs_softdep.c:6760]: Maybe offset is a better name for this?

File: sys/ufs/ffs/ffs_softdep.c [hunk 2]
```diff
@@ -6754,14 +6755,13 @@ complete_trunc_indir(freework)
...
 	else
-		start = sizeof(ufs2_daddr_t);
-	start *= freework->fw_start;
-	count = indirdep->ir_savebp->b_bcount - start;
-	start += (uintptr_t)indirdep->ir_savebp->b_data;
-	bzero((char *)start, count);
```
[the remark this answers is not in the document]
Comment (qwattash) [reply sys/ufs/ffs/ffs_softdep.c:6764]: Indeed, it's not even easier to read.

[the remark this answers is not in the document]
Comment (jrtc27) [reply sys/ufs/ffs/ffs_softdep.c:6764]: Yeah, new code is much clearer, hardest part was convincing myself the old code was doing what the new code does

[the remark this answers is not in the document]
Comment (brooksdavis) [reply sys/ufs/ffs/ffs_softdep.c:6764]: Probably worth dropping this in phabricator for Kirk while you've looked at it recently.

**Review by brooksdavis — APPROVED** [review]

**Pull request merged**